plugins/woocommerce/src/Blocks/BlockTypes/ProductGalleryLargeImageNextPrevious.php

```diff
@@ -55,6 +55,12 @@ protected function render( $attributes, $content, $block ) {
 			return null;
 		}
 
+		$featured_image_id = $product->get_image_id();
+
+		if( $featured_image_id && count( $product_gallery) === 1 && (int) $product_gallery[0] === (int) $featured_image_id ) {
+			return null;
+		}
+
```

Comment:
While it does address the case that was raised I think we could provide more generic solution to the problem. Thumbnails react correctly to the number of images as the case of overlap between featured image and gallery images is solved in utils used in other blocks.

I suggest we go this way and utilise the util so all the inner blocks share the same logic of getting images.

This is suggestion to replace lines 52-63 (GitHub doesn't allow marking suggestions to unchanged lines)

```suggestion
		$product_gallery_thumbnails_data = ProductGalleryUtils::get_product_gallery_image_data( $product );
		$product_gallery_images          = $product_gallery_thumbnails_data['images'];
		// Don't show the arrows block if there is only one image.
		if ( count( $product_gallery_images ) <= 1 ) {
			return '';
		}
```

`get_product_gallery_image_data` does all the things to return unique visible images so that would be a nicer solution 🙌

Reply:
@kmanijak, I agree, the suggested solution is better. I've implemented it.